src/blocks/weather/open_weather_map.rs

```diff
@@ -79,24 +94,198 @@ struct ApiWeather {
     description: String,
 }
 
-#[async_trait]
-impl WeatherProvider for Service<'_> {
-    async fn get_weather(&self, autolocated: Option<Coordinates>) -> Result<WeatherResult> {
-        let api_key = self.config.api_key.as_ref().or_error(|| {
-            format!("missing key 'service.api_key' and environment variable {API_KEY_ENV}",)
-        })?;
+#[derive(Deserialize, Debug)]
+struct City {
+    //This should be an u32, but the source data has a few errors
+    id: f32,
+    coord: CityCoord,
+}
+#[derive(Deserialize, Debug)]
+struct CityCoord {
+    lat: f64,
+    lon: f64,
+}
+
+static LOCATION_CACHE: Mutex<Option<String>> = Mutex::new(None);
```

Comment:
Does this prevent from having two blocks which show weather in different places?

Reply:
Yes... Good catch.